consent-ui/src/app/ais/enter-pin-page/enter-pin-page.component.ts

```diff
@@ -0,0 +1,48 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { StubUtil } from '../common/stub-util';
+import { ApiHeaders } from '../../api/api.headers';
+import { ConsentAuthorizationService } from '../../api';
+import { SessionService } from '../../common/session.service';
+
+@Component({
+  selector: 'consent-app-enter-pin-page',
+  templateUrl: './enter-pin-page.component.html',
+  styleUrls: ['./enter-pin-page.component.scss']
+})
+export class EnterPinPageComponent implements OnInit {
```

Comment:
To me, it makes sense to extract the PinInput component and use it in both AIS and PIS (like you did with EnterScaComponent). And in this case, `wrongPassword` will be `@Input` parameter

Reply:
I wasn't sure. Do ais and pis have the same endpoint for Pininput?

Reply:
Same as of now, you may add some `@Input` for submission function to be extremely generic

Reply:
No you are actually in the ais module. The `@Input wrongPassword ` ist actually in common.

Reply:
What I mean is to provide
```
 this.consentAuthorizationService
      .embeddedUsingPOST(
        this.authorizationSessionId,
        StubUtil.X_REQUEST_ID, // TODO: real values instead of stubs
        StubUtil.X_XSRF_TOKEN, // TODO: real values instead of stubs
        this.redirectCode,
        { scaAuthenticationData: { PSU_PASSWORD: pin } },
        'response'
      )
      .subscribe(res => {
        // redirect to the provided location
        this.sessionService.setRedirectCode(this.authorizationSessionId, res.headers.get(ApiHeaders.REDIRECT_CODE));
        console.log('REDIRECTING TO: ' + res.headers.get(ApiHeaders.LOCATION));
        window.location.href = res.headers.get(ApiHeaders.LOCATION);
      });
```
as Input to support different endpoints